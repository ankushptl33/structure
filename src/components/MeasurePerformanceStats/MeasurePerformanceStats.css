.measures__data--grid {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding-left: 35px;
  border-left: 1px solid #dedddd;
}
.measures__data--grid .measures__data--label {
  font-size: 14px;
  letter-spacing: 0.5px;
  color: gray;
  padding-left: 20px;
}
.measures__data--grid .measures__data--percent {
  padding-left: 20px;
  color: #515151;
}
