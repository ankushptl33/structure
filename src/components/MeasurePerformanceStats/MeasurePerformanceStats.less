.measures__data--grid {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding-left: 35px;
    border-left: 1px solid #DEDDDD;
    .measures__data--label {
        font-size: 14px;
        letter-spacing: 0.5px;
        color: gray;
        padding-left: 20px;
    }
    .measures__data--percent{
        padding-left: 20px; 
        color: #515151;
    }
}
    .measures__icons--grid{
        .measures__data--icons{
            display: flex;
            justify-content: center;
            align-items: center;
            .measures__data--arrow-up{
                font-size: 2rem;
            } 
            .measures__data--thumbs-up{
                font-size: 2rem;
                padding-left: 10px;
            } 
        }
    } 
