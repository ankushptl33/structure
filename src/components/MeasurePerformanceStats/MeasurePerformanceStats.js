import React from 'react';
import { Grid, Tooltip } from '@material-ui/core';
import './MeasurePerformanceStats.less';
import Typography from '@material-ui/core/Typography';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const MeasurePerformanceStats = props => {
    const {
        configuration,
        measurePerformance,
        isinverse,
        isleftfloated,
        colorCode
    } = props.measureData;

    let arrow, colorArrow, arrowTitle;
    let leftFloatedMeasures = null;
    let rightFloatedMeasures = null;

    if (isinverse) {
        arrow = configuration.iconArrowDown ? configuration.iconArrowDown : ['fal', 'arrow-up'];
        arrowTitle = 'Lower score for this measure is better';
    } else {
        arrow = configuration.iconArrowUp ? configuration.iconArrowUp : ['fal', 'arrow-down'];
        arrowTitle = 'Higher score for this measure is better';
    }

    if (isinverse) {
        colorArrow = configuration.colorArrowDown ? configuration.colorArrowDown : 'green';
    } else {
        colorArrow = configuration.colorArrowUp ? configuration.colorArrowUp : 'green';
    }

    const measures = measurePerformance.map((measuresData, index) => {
        return (
            <Grid key={index} item className='measures__data--grid'>
                <Typography variant='h6' className='measures__data--label'>
                    {measuresData.label}
                </Typography>
                <Typography variant='h4' className='measures__data--percent'>
                    {measuresData.data}%
                </Typography>
            </Grid>
        );
    });

    if (isleftfloated) {
        leftFloatedMeasures = measures;
    } else {
        rightFloatedMeasures = measures;
    }

    return (
        <Grid container spacing={16} className='measures__icons--grid'>
            {rightFloatedMeasures}
            <Grid item xs className='measures__data--icons'>
                <span className='measures__data--arrow-up'>
                    <Tooltip title={arrowTitle}>
                        {<FontAwesomeIcon icon={arrow} size='lg' color={colorArrow} />}
                    </Tooltip>
                </span>
                <span className={`measures__data--thumbs-up ${colorCode}`}>
                    {
                        <FontAwesomeIcon
                            icon={['fal', 'thumbs-up']}
                            size='lg'
                            color={configuration.colorThumbsUp}
                        />
                    }
                </span>
            </Grid>
            {leftFloatedMeasures}
        </Grid>
    );
};

MeasurePerformanceStats.defaultProps = {
    measureData: {
        measurePerformance: [
            {
                label: 'Achieved Performance',
                data: '88.08'
            },
            {
                label: 'CMS Benchmark',
                data: '45'
            },
            {
                label: 'Registry Average',
                data: '85'
            }
        ],
        isinverse: true,
        isleftfloated: false,
        configuration: {
            iconArrowUp: ['fal', 'arrow-up'],
            iconArrowDown: ['fal', 'arrow-down'],
            colorArrowUp: 'green',
            colorArrowDown: 'green',
            iconThumbsUp: ['fal', 'thumbs-up'],
            colorThumbsUp: 'green'
        }
    }
};
export default MeasurePerformanceStats;
