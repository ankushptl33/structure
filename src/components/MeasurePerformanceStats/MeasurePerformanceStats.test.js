import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MeasurePerformanceStats from '../../components/MeasurePerformanceStats/MeasurePerformanceStats';
import MeasurePropsData from '../../json/MeasurePerformanceStats.json';

Enzyme.configure({ adapter: new Adapter() });

describe('<MeasurePerformanceStats />', () => {
    test('renders', () => {
        const component = shallow(
            <MeasurePerformanceStats measureData={MeasurePropsData.measureData} />
        );
        expect(component).toHaveLength(1);
        expect(component).toMatchSnapshot();
    });

    it('will check props', () => {
        const component = shallow(
            <MeasurePerformanceStats measureData={MeasurePropsData.measureData} />
        ).props();
        expect(component).toBeDefined();
    });

    it('mount component with default props', () => {
        const component = shallow(<MeasurePerformanceStats />);
        expect(component.exists()).toBe(true);
    });
});
