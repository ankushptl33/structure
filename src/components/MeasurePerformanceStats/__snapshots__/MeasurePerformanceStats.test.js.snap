// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeasurePerformanceStats /> renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MeasurePerformanceStats
    measureData={
      Object {
        "isinverse": false,
        "isleftfloated": true,
        "measurePerformance": Array [
          Object {
            "data": "78.08",
            "label": "Achieved Performance",
          },
          Object {
            "data": "45",
            "label": "CMS Benchmark",
          },
          Object {
            "data": "85",
            "label": "Registry Average",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        null,
        <Grid
          className="singleGrid"
        >
          <Icon
            as="i"
            color="green"
            name="arrow up"
            size="huge"
          />
          <Icon
            as="i"
            color="green"
            name="thumbs up outline"
            size="huge"
          />
        </Grid>,
        Array [
          <Grid
            className="singleGrid"
          >
            <Statistic>
              <p>
                Achieved Performance
              </p>
              <StatisticValue
                text={true}
              >
                78.08
                %
              </StatisticValue>
            </Statistic>
          </Grid>,
          <Grid
            className="singleGrid"
          >
            <Statistic>
              <p>
                CMS Benchmark
              </p>
              <StatisticValue
                text={true}
              >
                45
                %
              </StatisticValue>
            </Statistic>
          </Grid>,
          <Grid
            className="singleGrid"
          >
            <Statistic>
              <p>
                Registry Average
              </p>
              <StatisticValue
                text={true}
              >
                85
                %
              </StatisticValue>
            </Statistic>
          </Grid>,
        ],
      ],
      "className": "gridCollection",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Icon
              as="i"
              color="green"
              name="arrow up"
              size="huge"
            />,
            <Icon
              as="i"
              color="green"
              name="thumbs up outline"
              size="huge"
            />,
          ],
          "className": "singleGrid",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "i",
              "color": "green",
              "name": "arrow up",
              "size": "huge",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "i",
              "color": "green",
              "name": "thumbs up outline",
              "size": "huge",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "children": <Statistic>
            <p>
              Achieved Performance
            </p>
            <StatisticValue
              text={true}
            >
              78.08
              %
            </StatisticValue>
          </Statistic>,
          "className": "singleGrid",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p>
                Achieved Performance
              </p>,
              <StatisticValue
                text={true}
              >
                78.08
                %
              </StatisticValue>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Achieved Performance",
              },
              "ref": null,
              "rendered": "Achieved Performance",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "78.08",
                  "%",
                ],
                "text": true,
              },
              "ref": null,
              "rendered": Array [
                "78.08",
                "%",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "children": <Statistic>
            <p>
              CMS Benchmark
            </p>
            <StatisticValue
              text={true}
            >
              45
              %
            </StatisticValue>
          </Statistic>,
          "className": "singleGrid",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p>
                CMS Benchmark
              </p>,
              <StatisticValue
                text={true}
              >
                45
                %
              </StatisticValue>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CMS Benchmark",
              },
              "ref": null,
              "rendered": "CMS Benchmark",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "45",
                  "%",
                ],
                "text": true,
              },
              "ref": null,
              "rendered": Array [
                "45",
                "%",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "children": <Statistic>
            <p>
              Registry Average
            </p>
            <StatisticValue
              text={true}
            >
              85
              %
            </StatisticValue>
          </Statistic>,
          "className": "singleGrid",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p>
                Registry Average
              </p>,
              <StatisticValue
                text={true}
              >
                85
                %
              </StatisticValue>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Registry Average",
              },
              "ref": null,
              "rendered": "Registry Average",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "85",
                  "%",
                ],
                "text": true,
              },
              "ref": null,
              "rendered": Array [
                "85",
                "%",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          null,
          <Grid
            className="singleGrid"
          >
            <Icon
              as="i"
              color="green"
              name="arrow up"
              size="huge"
            />
            <Icon
              as="i"
              color="green"
              name="thumbs up outline"
              size="huge"
            />
          </Grid>,
          Array [
            <Grid
              className="singleGrid"
            >
              <Statistic>
                <p>
                  Achieved Performance
                </p>
                <StatisticValue
                  text={true}
                >
                  78.08
                  %
                </StatisticValue>
              </Statistic>
            </Grid>,
            <Grid
              className="singleGrid"
            >
              <Statistic>
                <p>
                  CMS Benchmark
                </p>
                <StatisticValue
                  text={true}
                >
                  45
                  %
                </StatisticValue>
              </Statistic>
            </Grid>,
            <Grid
              className="singleGrid"
            >
              <Statistic>
                <p>
                  Registry Average
                </p>
                <StatisticValue
                  text={true}
                >
                  85
                  %
                </StatisticValue>
              </Statistic>
            </Grid>,
          ],
        ],
        "className": "gridCollection",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Icon
                as="i"
                color="green"
                name="arrow up"
                size="huge"
              />,
              <Icon
                as="i"
                color="green"
                name="thumbs up outline"
                size="huge"
              />,
            ],
            "className": "singleGrid",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "i",
                "color": "green",
                "name": "arrow up",
                "size": "huge",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "i",
                "color": "green",
                "name": "thumbs up outline",
                "size": "huge",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "children": <Statistic>
              <p>
                Achieved Performance
              </p>
              <StatisticValue
                text={true}
              >
                78.08
                %
              </StatisticValue>
            </Statistic>,
            "className": "singleGrid",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <p>
                  Achieved Performance
                </p>,
                <StatisticValue
                  text={true}
                >
                  78.08
                  %
                </StatisticValue>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Achieved Performance",
                },
                "ref": null,
                "rendered": "Achieved Performance",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "78.08",
                    "%",
                  ],
                  "text": true,
                },
                "ref": null,
                "rendered": Array [
                  "78.08",
                  "%",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "children": <Statistic>
              <p>
                CMS Benchmark
              </p>
              <StatisticValue
                text={true}
              >
                45
                %
              </StatisticValue>
            </Statistic>,
            "className": "singleGrid",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <p>
                  CMS Benchmark
                </p>,
                <StatisticValue
                  text={true}
                >
                  45
                  %
                </StatisticValue>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "CMS Benchmark",
                },
                "ref": null,
                "rendered": "CMS Benchmark",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "45",
                    "%",
                  ],
                  "text": true,
                },
                "ref": null,
                "rendered": Array [
                  "45",
                  "%",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "children": <Statistic>
              <p>
                Registry Average
              </p>
              <StatisticValue
                text={true}
              >
                85
                %
              </StatisticValue>
            </Statistic>,
            "className": "singleGrid",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <p>
                  Registry Average
                </p>,
                <StatisticValue
                  text={true}
                >
                  85
                  %
                </StatisticValue>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Registry Average",
                },
                "ref": null,
                "rendered": "Registry Average",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "85",
                    "%",
                  ],
                  "text": true,
                },
                "ref": null,
                "rendered": Array [
                  "85",
                  "%",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
